test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTransactionContextTest.java

```diff
@@ -131,6 +135,194 @@ public void customConflictObjectsNoConflictNoAbort()
                 .assertDoesNotThrow(TransactionAbortedException.class);
     }
 
+    @Data
+    @AllArgsConstructor
+    static class CustomSameHashConflictObject {
+        final String k1;
+        final String k2;
+    }
+
+    /** This test generates a custom object which has been registered
+     * with the serializer to always conflict, as it always hashes to
+     * and empty byte array.
+     */
+    @Test
+    public void customSameHashAlwaysConflicts() {
+        // Register a custom hasher which always hashes to an empty byte array
+        Serializers.JSON.registerCustomHasher(CustomSameHashConflictObject.class,
+                o -> new byte[0]);
+
+        CustomSameHashConflictObject c1 = new CustomSameHashConflictObject("a", "a");
```

Comment:
No, they are just internal keys that have no meaning, other than being equal. They are not related to the stream or map at all.